src/components/Api.js

```diff
@@ -0,0 +1,82 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _renderResult(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getUserInfoAndCards() {
+    return Promise.all([this.getInitialCards(), this.getUserInfo()]);
+  }
+
+  async getInitialCards() {
+    const res = await fetch(`${this._baseUrl}/cards`, {
+      method: "GET",
+      headers: this._headers,
+    });
+    return this._renderResult(res);
+  }
+
+  async addCard({ name, link }) {
+    const res = await fetch(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({ name, link }),
+    });
+    return this._renderResult(res);
+  }
+
+  async getUserInfo() {
+    const res = await fetch(`${this._baseUrl}/users/me`, {
+      method: "GET",
+      headers: this._headers,
+    });
+    return this._renderResult(res);
+  }
+
+  setUserInfo(name, about) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({ name: name, about: about }),
+    });
+  }
+
+  deleteCard(cardId) {
+    return fetch(`${this._baseUrl}/cards/${cardId}`, {
+      method: "DELETE",
+      headers: this._headers,
+    }).then(this.renderResult);
```

Comment:
`renderResult` doesn't exit. You have `_renderResult`

`    }).then(this._renderResult);
`

Reply:
* This remark hasn't been corrected https://github.com/mowooje/se_project_aroundtheus/pull/4#discussion_r1792812282 


Please, add `_` before the variable because you have `_renderResult`

Reply:
Just added it to the code